cmd/github-labels/github-labels.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Description: Generate the combined GitHub labels database for the
+#   specified repository.
+
+set -e
+
+script_name=${0##*/}
+
+source "/etc/os-release" || "source /usr/lib/os-release"
+
+self_dir=$(dirname "$(readlink -f "$0")")
+cidir="${self_dir}/../../.ci"
+source "${cidir}/lib.sh"
+
+typeset -r labels_file="labels.yaml"
+typeset -r labels_template="${labels_file}.in"
+
+typeset -r master_labels_file="${self_dir}/${labels_file}"
+typeset -r master_labels_template="${self_dir}/${labels_template}"
+
+# The GitHub labels API requires a colour for each label so
+# default to a white background.
+typeset -r default_color="ffffff"
+
+merge_yaml()
+{
+	local -r file1="$1"
+	local -r file2="$2"
+	local -r out="$3"
+
+	[ -n "$file1" ] || die "need 1st file"
+	[ -n "$file2" ] || die "need 2nd file"
+	[ -n "$out" ] || die "need output file"
+
+	[ -z "$(command -v yq)" ] && install_yq
+
+	yq merge "$file1" --append "$file2" > "$out"
+}
+
+check_yaml()
+{
+	local -r file="$1"
+
+	[ -n "$file" ] || die "need file to check"
+
+	yq read "$file" >/dev/null
+
+	yamllint --strict "$file"
```

Comment:
above we check if we have `yq` installed - should we do similar (but maybe have to abort) if we don't find `yamllint`?

Reply:
Fixed - I want to keep this PR as minimal as possible so it just dies if `yamllint` isn't installed for now  - it will be installed on the CI systems.